### PR TITLE
[simd/v3i]: Implement v128 replace_lane

### PR DESCRIPTION
**Pull request by haoyu-zc**
Tested by:
` make v3i -j && bin/spectest.v3i -tk test/regress/simd/v128_replace_lane.bin.wast`

### CODE REVIEW

File: src/engine/v3/V3Interpreter.v3 [hunk 1]
```diff
@@ -1139,6 +1145,28 @@ component V3Interpreter {
 		}
 		push(Value.V128(low, high));
 	}
+	def doReplaceLane<T, S>(log2_size: u3, view: S -> T, unbox: Value -> S) {
```
Comment (titzer) [inline src/engine/v3/V3Interpreter.v3:1148]: You could also pass one of `popi`, `popf`, etc, here.

Comment (haoyu-zc) [reply src/engine/v3/V3Interpreter.v3:1148]: Ah yes that's a miss. Updating it.

**Comment by haoyu-zc** [comment]
Updated. Also removed some dead code.

**Review by titzer — APPROVED** [review]
lgtm

**Pull request merged**